tools/pubsys/src/aws/validate_ssm/mod.rs

```diff
@@ -186,87 +151,68 @@ pub(crate) fn validate_parameters_in_region(
+/// value as `String`.
+pub(crate) async fn parse_expected_parameters(
+    expected_parameters_file: &PathBuf,
+) -> Result<HashMap<Region, HashMap<SsmKey, String>>> {
+    // Parse the JSON file as a HashMap of region_name, mapped to a HashMap of parameter_name and
+    // parameter_value
+    let expected_parameters: HashMap<RegionName, HashMap<ParameterName, ParameterValue>> =
+        serde_json::from_reader(&File::open(expected_parameters_file.clone()).context(
+            error::ReadExpectedParameterFileSnafu {
+                path: expected_parameters_file,
             },
         )?)
-        .context(error::ParseExpectedParameterListSnafu)?;
-
-        // Iterate over the parsed HashMap, converting the nested HashMap into a HashMap of Region
-        // mapped to a HashMap of SsmKey, SsmValue
-        parameter_list
-            .iter()
-            .filter(|(region, _)| validation_regions.contains(region))
-            .flat_map(|(region, ami_ids)| {
-                ami_ids
-                    .iter()
-                    .map(move |(ami_id, param_names)| (region, ami_id, param_names))
-            })
-            .flat_map(|(region, ami_id, params)| {
-                params.iter().map(move |(parameter_name, parameter_value)| {
-                    (
-                        region.clone(),
-                        ami_id.clone(),
-                        parameter_name.clone(),
-                        parameter_value.clone(),
-                    )
-                })
-            })
-            .for_each(|(region, ami_id, parameter_name, parameter_value)| {
-                parameter_map
-                    .entry(Region::new(region.clone()))
-                    .or_insert(HashMap::new())
-                    .insert(
-                        SsmKey::new(Region::new(region), parameter_name),
-                        SsmValue {
-                            value: parameter_value,
-                            ami_id,
-                        },
-                    );
-            });
-    }
+        .context(error::ParseExpectedParameterFileSnafu)?;
+
+    // Iterate over the parsed HashMap, converting the nested HashMap into a HashMap of Region
+    // mapped to a HashMap of SsmKey, String
+    let parameter_map = expected_parameters
+        .into_iter()
+        .map(|(region, parameters)| {
+            (
+                Region::new(region.clone()),
+                parameters
+                    .into_iter()
+                    .map(|(parameter_name, parameter_value)| {
+                        (
+                            SsmKey::new(Region::new(region.clone()), parameter_name),
+                            parameter_value,
+                        )
+                    })
+                    .collect::<HashMap<SsmKey, String>>(),
+            )
+        })
+        .collect::<HashMap<Region, HashMap<SsmKey, String>>>();
```

Comment:
Just a tip, I think this code is fine, but you could probably remove this turbofish. The compiler should be able to deduce the type from the return type of the function.

```suggestion
        .collect();
```